model_access_keys.go

```diff
@@ -69,8 +69,27 @@ func getAccessKey(ctx context.Context, id string, opts ...ModelOps) (*AccessKey,
 	return key, nil
 }
 
-// getAccessKeys will get all the access keys that match the metadata search
-func getAccessKeys(ctx context.Context, xPubID string, metadata *Metadata, conditions *map[string]interface{},
+// getAccessKeys will get all the access keys with the given conditions
+func getAccessKeys(ctx context.Context, metadata *Metadata, conditions *map[string]interface{},
+	queryParams *datastore.QueryParams, opts ...ModelOps) ([]*AccessKey, error) {
+
+	modelItems := make([]*AccessKey, 0)
+	if err := getModelsByConditions(ctx, ModelAccessKey, &modelItems, metadata, conditions, queryParams, opts); err != nil {
```

Comment:
Does opts need the `...`

`opts` vs `opts...`

Not sure if that is an error or not...

Reply:
ModelOps was refactored by Goland into `[]ModelOps` which is the reason this works. I changed it to follow the rest of the actions.